streams/src/test/java/org/apache/kafka/streams/kstream/internals/KGroupedTableImplTest.java

```diff
@@ -199,7 +199,7 @@ public void shouldReduceAndMaterializeResults() {
         final Map<String, Integer> results = getReducedResults(reduced);
         try (final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), props)) {
             assertReduced(results, topic, driver);
-            final KeyValueStore<String, Integer> reduce = (KeyValueStore<String, Integer>) driver.getStateStore("reduce");
+            final KeyValueStore<String, Integer> reduce = driver.getKeyValueStore("reduce");
```

Comment:
This is an interesting issue and I am wondering who we could address it... Any thoughts?

Reply:
What issue are you referring to?

Reply:
I see.

I'm wondering if we really need the store facade: as I mentioned in another comment, if we only want to allow `puts` before the starting of the topology, but not after records being processed, then just setting the record context in `TopologyTestDriver` constructor should be sufficient.